pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyLocale.java

```diff
@@ -0,0 +1,18 @@
+package com.redhat.pantheon.model.assembly;
+
+import com.redhat.pantheon.model.api.Child;
+import com.redhat.pantheon.model.api.annotation.JcrPrimaryType;
+import com.redhat.pantheon.model.module.ModuleLocale;
+import com.redhat.pantheon.model.module.SourceContent;
+
+/**
+ * A specific assembly locale node which houses asciidoc source and variants.
+ * A locale will contain two folders to store the source content (draft and released),
+ * and the different variants (the rendered content)
+ */
+@JcrPrimaryType("pant:assemblyLocale")
+public interface AssemblyLocale extends ModuleLocale {
```

Comment:
So I agree with the first part - I'm not sure that it makes sense for any Assembly classes to extend the Module classes.

However, I'm not sold on the second idea for ContentLocale. I do like the idea of having separate node types, because otherwise we could accidentally "jump the tracks" in the code, like this example that I'll make up:

```
Module m;
m.getContentLocale("en-US").getAsmVariant("rhel-8").getReleased().getPage(3);
```

Notice that we started with a Module object but somehow wound up with a call to getPage which only makes sense in an assembly context - but the code could be syntactically correct and could compile (if not run successfully) because the ContentLocale object would need to have methods for both assemblies and modules if we don't break them out into separate nodes/classes.

Reply:
I was discussing with Lisa the trade-offs of trying to remove some of the code duplication that will eventually come out of this (we happen to have two **very** **similar** content structures) and in the end what we realized is that it's probably better to endure the duplication now and revisit the need for it down the line.

Reply:
Yeah, I would agree with that. Let's keep things separate until the code solidifies and then we can re-evaluate based on where things actually wound up.